src/dft/backends/mklcpu/CMakeLists.txt

```diff
@@ -60,7 +59,7 @@ endif()
 set_target_properties(${LIB_OBJ} PROPERTIES
   POSITION_INDEPENDENT_CODE ON
 )
-target_link_libraries(${LIB_NAME} PUBLIC ${LIB_OBJ})
+target_link_libraries(${LIB_NAME} PUBLIC ${LIB_OBJ} onemkl_dft)
```

Comment:
From my current understanding of the CMake `onemkl_dft` should be linked to the tests and examples only to match with the other domains.

Reply:
Sorry, I'm having trouble seeing where this happens in the project. Saying that, I also don't see how the other domains are linking the runtime stuff at all.

Reply:
I think that RNG might be doing their runtime dispatch in the headers

Reply:
You're right, my bad. That looks good to me so let's keep it here!

Reply:
I think the `onemkl_${domain}` libraries may not be linked at all but opened in `dlopen` instead.

Reply:
I see! I think I know what I need to do but this is a little awkward with templates :D